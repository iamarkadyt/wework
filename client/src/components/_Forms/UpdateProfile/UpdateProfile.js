import React from 'react';
import { connect } from 'react-redux'
import './UpdateProfile.scss'

import Field from '../../Field/Field'
import Overlay from '../../Overlay/Overlay'
import { updateUsersProfile } from '../../../state/actions/profileActions'

class UpdateProfile extends React.Component {
    initialState = {
        showSocial: false,
        creatingProfile: true
    }

    constructor(props) {
        super(props)
        const { profile } = props

        if (profile) {
            this.state = {
                ...this.initialState,
                creatingProfile: false,
                ...profile
            }
        } else {
            this.state = this.initialState
        }
    }

    handleDismiss = () => {
        const { history } = this.props
        history.goBack()
    }

    render() {
        const { updateUsersProfile, errors } = this.props

        return (
            <Overlay onBackdropClick={() => this.handleDismiss()}>
                <form className="UpdateProfile-container" onSubmit={e => {
                    e.preventDefault()
                    const { showSocial, creatingProfile, ...data } = this.state
                    updateUsersProfile(data, () => this.handleDismiss())
                }}>
                    <h1>{this.state.creatingProfile ? 'Create' : 'Update'} profile</h1>
                    <Field
                        type="text"
                        name="handle"
                        label="Profile handle:"
                        value={this.state.handle}
                        onChange={e => this.setState({ handle: e.target.value })}
                        error={errors.handle}
                        placeholder="someperson123" />
                    <Field
                        type="text"
                        name="company"
                        label="Company:"
                        value={this.state.company}
                        onChange={e => this.setState({ company: e.target.value })}
                        error={errors.company}
                        placeholder="Facebook" />
                    <Field
                        type="text"
                        name="website"
                        label="Personal Website:"
                        value={this.state.website}
                        onChange={e => this.setState({ website: e.target.value })}
                        error={errors.website}
                        placeholder="http://mywebsite.com" />
                    <Field
                        type="text"
                        name="location"
                        label="Location:"
                        value={this.state.location}
                        onChange={e => this.setState({ location: e.target.value })}
                        error={errors.location}
                        placeholder="Seattle, WA" />
                    <Field
                        type="text"
                        name="title"
                        label="Job title:"
                        value={this.state.title}
                        onChange={e => this.setState({ title: e.target.value })}
                        error={errors.title}
                        placeholder="Marketing Analyst" />
                    <Field
                        type="list"
                        name="status"
                        label="Job seeker status:"
                        list={['Not open for Employment', 'Available for Employment', 'Actively Seeking']}
                        value={this.state.status}
                        onChange={value => this.setState({ status: value })}
                        error={errors.status} />
                    <Field
                        type="multiselect"
                        name="skills"
                        label="Skills:"
                        list={['Python', 'Java', 'C++', 'Hospitality', 'Whatnot else']}
                        value={this.state.skills}
                        onChange={value => this.setState({ skills: value })}
                        error={errors.skills} />
                    <Field
                        type="textarea"
                        name="bio"
                        label="Bio:"
                        value={this.state.bio}
                        onChange={e => this.setState({ bio: e.target.value })}
                        error={errors.bio}
                        placeholder="It's okay to brag here.." />
                    <Field
                        type="button"
                        label="Toggle social media section"
                        onClick={e => {
                            e.preventDefault()
                            this.setState(prevState => ({ showSocial: !prevState.showSocial }))
                        }} />
                    <div style={{ display: this.state.showSocial ? 'block' : 'none' }}>
                        <Field
                            type="text"
                            name="youtube"
                            label="Youtube:"
                            value={this.state.youtube}
                            onChange={e => this.setState({ youtube: e.target.value })}
                            error={errors.youtube}
                            placeholder="Please enter full channel URL" />
                        <Field
                            type="text"
                            name="twitter"
                            label="Twitter:"
                            value={this.state.twitter}
                            onChange={e => this.setState({ twitter: e.target.value })}
                            error={errors.twitter}
                            placeholder="Twitter profile URL" />
                        <Field
                            type="text"
                            name="instagram"
                            label="Instagram:"
                            value={this.state.instagram}
                            onChange={e => this.setState({ instagram: e.target.value })}
                            error={errors.instagram}
                            placeholder="Instagram URL" />
                        <Field
                            type="text"
                            name="facebook"
                            label="Facebook:"
                            value={this.state.facebook}
                            onChange={e => this.setState({ facebook: e.target.value })}
                            error={errors.facebook}
                            placeholder="Facebook profile URL" />
                        <Field
                            type="text"
                            name="linkedin"
                            label="LinkedIn:"
                            value={this.state.linkedin}
                            onChange={e => this.setState({ linkedin: e.target.value })}
                            error={errors.linkedin}
                            placeholder="LinkedIn URL" />
                        <Field
                            type="text"
                            name="githubusername"
                            label="Github:"
                            value={this.state.githubusername}
                            onChange={e => this.setState({ githubusername: e.target.value })}
                            error={errors.githubusername}
                            placeholder="Github username only" />
                    </div>
                    <Field
                        type="submit"
                        label={this.state.creatingProfile ? 'Create' : 'Update'} />
                    <Field
                        type="button"
                        label="Cancel"
                        onClick={e => {
                            e.preventDefault()
                            this.handleDismiss()
                        }} />
                </form>
            </Overlay>
        )
    }
}

export default connect(state => ({
    errors: state.err.formErrors
}), { updateUsersProfile })(UpdateProfile)
