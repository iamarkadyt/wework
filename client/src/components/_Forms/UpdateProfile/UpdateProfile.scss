form.UpdateProfile-container {
    width: 40rem;
    padding: 1.5rem;
}