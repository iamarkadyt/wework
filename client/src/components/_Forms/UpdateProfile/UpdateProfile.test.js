/**
 * Contract:
 *
 * What do I render?
 * (snapshot)
 * - Overlay
 *   - form
 *     - h1
 *     - 9 Fields
 *     - next 6 Fields hidden under a toggle
 *     - Submit and Cancel button Fields
 *
 * What do I pass components that I render?
 * (props passing)
 * - Look at all Fields,
 * - form
 * - Overlay
 * - h1
 * - div hiding social media fields
 *
 * Do props I receive affect what I render?
 * (props effects)
 * - errors object triggers error displaying
 * - if profile object isn't passed, UpdateProfile form turns into CreateProfile (h1, submit btn label)
 *
 * What do I do with functions (all of them)?
 * (functions / args passing)
 * - Look at:
 *   - updateUsersProfile
 *   - handleDismiss
 *   - preventDefault calls
 *   - history.push calls
 *   - setState calls
 *
 * Do I hold anything in state? HOW and WHEN do I update it? Invalidation? How does state affect other components?
 * (state)
 * - social media text Fields window state
 *   > controls abovementioned window
 *   > controlled through toggle button
 * - creatingProfile flag
 *   > conrols contents of h1 and submit button
 *   > controlled through profile object in props (undef > true, def > false)
 * - text fields states
 *
 * Any context used?
 * - redux: 1 thunk action, errors object from state
 *
 * Public ref API?
 * - n/a
 *
 * Lifecycle hooks side effects?
 * (lifecycle hooks)
 * - constructor:
 *   > if profile is passed, destructure it into state, set creatingProfile state flag to false
 *   > else apply initial state
 *
 * User interaction?
 * (interaction)
 * - Look at:
 *   - all Fields: typing, clicking, toggling etc.
 *   - form submission (correct/incorrect)
 *   - again, make sure preventDefault calls are present
 *   - cancel button click
 *   - backdrop click
 *
 */

import React from 'react'
import { shallow } from 'enzyme'
import { UpdateProfile } from './UpdateProfile'
import cloneDeep from 'lodash.clonedeep'

describe('UpdateProfile', () => {
  let props
  let mountedUpdateProfile

  const getMockProps = () => {
    return cloneDeep({
      errors: {},
      updateUsersProfile: jest.fn(),
      history: { 
        goBack: jest.fn() 
      }
    })
  }

  const comp = () => {
    if (!mountedUpdateProfile) {
      mountedUpdateProfile = shallow(<UpdateProfile {...props} />)
    }
    return mountedUpdateProfile
  }

  beforeEach(() => {
    props = getMockProps()
    mountedUpdateProfile = undefined
  })

  describe('props passing', () => {
    describe('handle Field', () => {
      it('receives correct props', () => {
        const expectedKeys = [
	  'type',
	  'name',
	  'label',
	  'value',
	  'onChange',
	  'error',
	  'placeholder'
        ]
        expect(Object.keys(comp().find('[name="handle"]').props())).toStrictEqual(expectedKeys)
      })

      it('receives undefined for [error] prop if props.errors.handle is undefined', () => {
        delete props.errors.handle
        expect(comp().find('[name="handle"]').prop('error')).not.toBeDefined()
      })

      it('receives error message for [error] props if props.errors.handle is defined', () => {
        const msg = 'something happened...'
        props.errors.handle = msg
        expect(comp().find('[name="handle"]').prop('error')).toBe(msg)
      })	

      it('receives correct [type] prop', () => {
        expect(comp().find('[name="handle"]').prop('type')).toBe('text')
      })

      it('receives correct [label] prop', () => {
        expect(comp().find('[name="handle"]').prop('label')).toBe('Profile handle:')
      })

      it('receives correct [placeholder] prop', () => {
        expect(comp().find('[name="handle"]').prop('placeholder')).toBe('someperson123')
      })

      it('receives undefined for [value] prop if state.handle is undefined', () => {
        comp().setState({ handle: undefined })
        expect(comp().find('[name="handle"]').prop('value')).not.toBeDefined()
      })

      it('receives state.handle for [value] prop if former is defined', () => {
        const state = 'some value'
        comp().setState({ handle: state })
        expect(comp().find('[name="handle"]').prop('value')).toBe(state)
      })

      it('receives function for [onChange] prop', () => {
        expect(comp().find('[name="handle"]').prop('onChange')).toEqual(expect.any(Function))
      })
    })
    
    describe('company Field', () => {
      it('receives correct props', () => {
        const expectedKeys = [
	  'type',
	  'name',
	  'label',
	  'value',
	  'onChange',
	  'error',
	  'placeholder'
        ]
        expect(Object.keys(comp().find('[name="company"]').props())).toStrictEqual(expectedKeys)
      })

      it('receives undefined for [error] prop if props.errors.company is undefined', () => {
        delete props.errors.company
        expect(comp().find('[name="company"]').prop('error')).not.toBeDefined()
      })

      it('receives error message for [error] props if props.errors.company is defined', () => {
        const msg = 'something happened...'
        props.errors.company = msg
        expect(comp().find('[name="company"]').prop('error')).toBe(msg)
      })	

      it('receives correct [type] prop', () => {
        expect(comp().find('[name="company"]').prop('type')).toBe('text')
      })

      it('receives correct [label] prop', () => {
        expect(comp().find('[name="company"]').prop('label')).toBe('Company:')
      })

      it('receives correct [placeholder] prop', () => {
        expect(comp().find('[name="company"]').prop('placeholder')).toBe('Company name')
      })

      it('receives undefined for [value] prop if state.company is undefined', () => {
        comp().setState({ company: undefined })
        expect(comp().find('[name="company"]').prop('value')).not.toBeDefined()
      })

      it('receives state.company for [value] prop if former is defined', () => {
        const state = 'some value'
        comp().setState({ company: state })
        expect(comp().find('[name="company"]').prop('value')).toBe(state)
      })

      it('receives function for [onChange] prop', () => {
        expect(comp().find('[name="company"]').prop('onChange')).toEqual(expect.any(Function))
      })
    })

    describe('website Field', () => {
      it('receives correct props', () => {
        const expectedKeys = [
	  'type',
	  'name',
	  'label',
	  'value',
	  'onChange',
	  'error',
	  'placeholder'
        ]
        expect(Object.keys(comp().find('[name="website"]').props())).toStrictEqual(expectedKeys)
      })

      it('receives undefined for [error] prop if props.errors.website is undefined', () => {
        delete props.errors.website
        expect(comp().find('[name="website"]').prop('error')).not.toBeDefined()
      })

      it('receives error message for [error] props if props.errors.website is defined', () => {
        const msg = 'something happened...'
        props.errors.website = msg
        expect(comp().find('[name="website"]').prop('error')).toBe(msg)
      })	

      it('receives correct [type] prop', () => {
        expect(comp().find('[name="website"]').prop('type')).toBe('text')
      })

      it('receives correct [label] prop', () => {
        expect(comp().find('[name="website"]').prop('label')).toBe('Personal Website:')
      })

      it('receives correct [placeholder] prop', () => {
        expect(comp().find('[name="website"]').prop('placeholder')).toBe('http://mywebsite.com')
      })

      it('receives undefined for [value] prop if state.website is undefined', () => {
        comp().setState({ website: undefined })
        expect(comp().find('[name="website"]').prop('value')).not.toBeDefined()
      })

      it('receives state.website for [value] prop if former is defined', () => {
        const state = 'some value'
        comp().setState({ website: state })
        expect(comp().find('[name="website"]').prop('value')).toBe(state)
      })

      it('receives function for [onChange] prop', () => {
        expect(comp().find('[name="website"]').prop('onChange')).toEqual(expect.any(Function))
      })
    })

    describe('location Field', () => {
      it('receives correct props', () => {
        const expectedKeys = [
	  'type',
	  'name',
	  'label',
	  'value',
	  'onChange',
	  'error',
	  'placeholder'
        ]
        expect(Object.keys(comp().find('[name="location"]').props())).toStrictEqual(expectedKeys)
      })

      it('receives undefined for [error] prop if props.errors.location is undefined', () => {
        delete props.errors.location
        expect(comp().find('[name="location"]').prop('error')).not.toBeDefined()
      })

      it('receives error message for [error] props if props.errors.location is defined', () => {
        const msg = 'something happened...'
        props.errors.location = msg
        expect(comp().find('[name="location"]').prop('error')).toBe(msg)
      })	

      it('receives correct [type] prop', () => {
        expect(comp().find('[name="location"]').prop('type')).toBe('text')
      })

      it('receives correct [label] prop', () => {
        expect(comp().find('[name="location"]').prop('label')).toBe('Location:')
      })

      it('receives correct [placeholder] prop', () => {
        expect(comp().find('[name="location"]').prop('placeholder')).toBe('Seattle, WA')
      })

      it('receives undefined for [value] prop if state.location is undefined', () => {
        comp().setState({ location: undefined })
        expect(comp().find('[name="location"]').prop('value')).not.toBeDefined()
      })

      it('receives state.location for [value] prop if former is defined', () => {
        const state = 'some value'
        comp().setState({ location: state })
        expect(comp().find('[name="location"]').prop('value')).toBe(state)
      })

      it('receives function for [onChange] prop', () => {
        expect(comp().find('[name="location"]').prop('onChange')).toEqual(expect.any(Function))
      })
    })

    describe('title Field', () => {
      it('receives correct props', () => {
        const expectedKeys = [
	  'type',
	  'name',
	  'label',
	  'value',
	  'onChange',
	  'error',
	  'placeholder'
        ]
        expect(Object.keys(comp().find('[name="title"]').props())).toStrictEqual(expectedKeys)
      })

      it('receives undefined for [error] prop if props.errors.title is undefined', () => {
        delete props.errors.title
        expect(comp().find('[name="title"]').prop('error')).not.toBeDefined()
      })

      it('receives error message for [error] props if props.errors.title is defined', () => {
        const msg = 'something happened...'
        props.errors.title = msg
        expect(comp().find('[name="title"]').prop('error')).toBe(msg)
      })	

      it('receives correct [type] prop', () => {
        expect(comp().find('[name="title"]').prop('type')).toBe('text')
      })

      it('receives correct [label] prop', () => {
        expect(comp().find('[name="title"]').prop('label')).toBe('Job title:')
      })

      it('receives correct [placeholder] prop', () => {
        expect(comp().find('[name="title"]').prop('placeholder')).toBe('Marketing Analyst')
      })

      it('receives undefined for [value] prop if state.title is undefined', () => {
        comp().setState({ title: undefined })
        expect(comp().find('[name="title"]').prop('value')).not.toBeDefined()
      })

      it('receives state.title for [value] prop if former is defined', () => {
        const state = 'some value'
        comp().setState({ title: state })
        expect(comp().find('[name="title"]').prop('value')).toBe(state)
      })

      it('receives function for [onChange] prop', () => {
        expect(comp().find('[name="title"]').prop('onChange')).toEqual(expect.any(Function))
      })
    })

    describe('status Field', () => {
      it('receives correct props', () => {
        const expectedKeys = [
	  'type',
	  'name',
	  'label',
	  'list',
	  'value',
	  'onChange',
	  'error'
        ]
        expect(Object.keys(comp().find('[name="status"]').props())).toStrictEqual(expectedKeys)
      })

      it('receives undefined for [error] prop if props.errors.status is undefined', () => {
        delete props.errors.status
        expect(comp().find('[name="status"]').prop('error')).not.toBeDefined()
      })

      it('receives error message for [error] props if props.errors.status is defined', () => {
        const msg = 'something happened...'
        props.errors.status = msg
        expect(comp().find('[name="status"]').prop('error')).toBe(msg)
      })	

      it('receives correct [type] prop', () => {
        expect(comp().find('[name="status"]').prop('type')).toBe('list')
      })

      it('receives correct [label] prop', () => {
        expect(comp().find('[name="status"]').prop('label')).toBe('Job seeker status:')
      })

      it('receives correct [list] prop', () => {
        expect(comp().find('[name="status"]').prop('list')).toEqual(expect.any(Array))
      })

      it('receives undefined for [value] prop if state.status is undefined', () => {
        comp().setState({ status: undefined })
        expect(comp().find('[name="status"]').prop('value')).not.toBeDefined()
      })

      it('receives state.status for [value] prop if former is defined', () => {
        const state = ['some value', 'some 2nd value']
        comp().setState({ status: state })
        expect(comp().find('[name="status"]').prop('value')).toBe(state)
      })

      it('receives function for [onChange] prop', () => {
        expect(comp().find('[name="status"]').prop('onChange')).toEqual(expect.any(Function))
      })
    })

    describe('skills Field', () => {
      it('receives correct props', () => {
        const expectedKeys = [
	  'type',
	  'name',
	  'label',
	  'list',
	  'value',
	  'onChange',
	  'error'
        ]
        expect(Object.keys(comp().find('[name="skills"]').props())).toStrictEqual(expectedKeys)
      })

      it('receives undefined for [error] prop if props.errors.skills is undefined', () => {
        delete props.errors.skills
        expect(comp().find('[name="skills"]').prop('error')).not.toBeDefined()
      })

      it('receives error message for [error] props if props.errors.skills is defined', () => {
        const msg = 'something happened...'
        props.errors.skills = msg
        expect(comp().find('[name="skills"]').prop('error')).toBe(msg)
      })	

      it('receives correct [type] prop', () => {
        expect(comp().find('[name="skills"]').prop('type')).toBe('multiselect')
      })

      it('receives correct [label] prop', () => {
        expect(comp().find('[name="skills"]').prop('label')).toBe('Skills:')
      })

      it('receives correct [list] prop', () => {
        expect(comp().find('[name="skills"]').prop('list')).toEqual(expect.any(Array))
      })

      it('receives undefined for [value] prop if state.skills is undefined', () => {
        comp().setState({ skills: undefined })
        expect(comp().find('[name="skills"]').prop('value')).not.toBeDefined()
      })

      it('receives state.skills for [value] prop if former is defined', () => {
        const state = ['some value', 'some 2nd value']
        comp().setState({ skills: state })
        expect(comp().find('[name="skills"]').prop('value')).toBe(state)
      })

      it('receives function for [onChange] prop', () => {
        expect(comp().find('[name="skills"]').prop('onChange')).toEqual(expect.any(Function))
      })
    })

    describe('bio Field', () => {
      it('receives correct props', () => {
        const expectedKeys = [
	  'type',
	  'name',
	  'label',
	  'value',
	  'onChange',
          'error',
          'placeholder'
        ]
        expect(Object.keys(comp().find('[name="bio"]').props())).toStrictEqual(expectedKeys)
      })

      it('receives undefined for [error] prop if props.errors.bio is undefined', () => {
        delete props.errors.bio
        expect(comp().find('[name="bio"]').prop('error')).not.toBeDefined()
      })

      it('receives error message for [error] props if props.errors.bio is defined', () => {
        const msg = 'something happened...'
        props.errors.bio = msg
        expect(comp().find('[name="bio"]').prop('error')).toBe(msg)
      })	

      it('receives correct [type] prop', () => {
        expect(comp().find('[name="bio"]').prop('type')).toBe('textarea')
      })

      it('receives correct [label] prop', () => {
        expect(comp().find('[name="bio"]').prop('label')).toBe('Bio:')
      })

      it('receives correct [placeholder] prop', () => {
        expect(comp().find('[name="bio"]').prop('placeholder')).toBe('It\'s okay to brag here..')
      })

      it('receives undefined for [value] prop if state.bio is undefined', () => {
        comp().setState({ bio: undefined })
        expect(comp().find('[name="bio"]').prop('value')).not.toBeDefined()
      })

      it('receives state.bio for [value] prop if former is defined', () => {
        const state = ['some value', 'some 2nd value']
        comp().setState({ bio: state })
        expect(comp().find('[name="bio"]').prop('value')).toBe(state)
      })

      it('receives function for [onChange] prop', () => {
        expect(comp().find('[name="bio"]').prop('onChange')).toEqual(expect.any(Function))
      })
    })

    describe('social media section toggle button', () => {
      it('receives correct props', () => {
        const expectedProps = [
          'type',
          'name',
          'label',
          'onClick'
        ]
        expect(Object.keys(comp().find('[name="toggle"]').props())).toStrictEqual(expectedProps)
      })

      it('receives correct [type] prop', () => {
        expect(comp().find('[name="toggle"]').prop('type')).toBe('button')
      })

      it('receives correct [label] prop', () => {
        expect(comp().find('[name="toggle"]').prop('label')).toBe('Toggle social media section')
      })

      it('receives correct [onClick] prop', () => {
        expect(comp().find('[name="toggle"]').prop('onClick')).toEqual(expect.any(Function))
      })
    })
  })

  describe('props effects', () => {
    it('', () => {})
  })

  describe('functions use', () => {
    it('', () => {})
  })

  describe('state', () => {
    it('', () => {})
  })

  describe('lifecycle', () => {
    it('', () => {})
  })

  describe('interaction', () => {
    it('', () => {})
  })

  it('passes a snapshot test', () => {
    expect(comp()).toMatchSnapshot()
  })
})
