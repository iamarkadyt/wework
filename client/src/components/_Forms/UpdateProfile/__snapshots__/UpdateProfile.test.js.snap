// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpdateProfile passes a snapshot test 1`] = `
<Overlay
  onBackdropClick={[Function]}
>
  <form
    className="UpdateProfile-container"
    onSubmit={[Function]}
  >
    <h1>
      Create
       profile
    </h1>
    <field
      label="Profile handle:"
      name="handle"
      onChange={[Function]}
      placeholder="someperson123"
      type="text"
    />
    <field
      label="Company:"
      name="company"
      onChange={[Function]}
      placeholder="Company name"
      type="text"
    />
    <field
      label="Personal Website:"
      name="website"
      onChange={[Function]}
      placeholder="http://mywebsite.com"
      type="text"
    />
    <field
      label="Location:"
      name="location"
      onChange={[Function]}
      placeholder="Seattle, WA"
      type="text"
    />
    <field
      label="Job title:"
      name="title"
      onChange={[Function]}
      placeholder="Marketing Analyst"
      type="text"
    />
    <field
      label="Job seeker status:"
      list={
        Array [
          "Not open for Employment",
          "Available for Employment",
          "Actively Seeking",
        ]
      }
      name="status"
      onChange={[Function]}
      type="list"
    />
    <field
      label="Skills:"
      list={
        Array [
          "Python",
          "Java",
          "C++",
          "Hospitality",
          "Whatnot else",
        ]
      }
      name="skills"
      onChange={[Function]}
      type="multiselect"
    />
    <field
      label="Bio:"
      name="bio"
      onChange={[Function]}
      placeholder="It's okay to brag here.."
      type="textarea"
    />
    <field
      label="Toggle social media section"
      name="toggle"
      onClick={[Function]}
      type="button"
    />
    <div
      style={
        Object {
          "display": "none",
        }
      }
    >
      <field
        label="Youtube:"
        name="youtube"
        onChange={[Function]}
        placeholder="Please enter full channel URL"
        type="text"
      />
      <field
        label="Twitter:"
        name="twitter"
        onChange={[Function]}
        placeholder="Twitter profile URL"
        type="text"
      />
      <field
        label="Instagram:"
        name="instagram"
        onChange={[Function]}
        placeholder="Instagram URL"
        type="text"
      />
      <field
        label="Facebook:"
        name="facebook"
        onChange={[Function]}
        placeholder="Facebook profile URL"
        type="text"
      />
      <field
        label="LinkedIn:"
        name="linkedin"
        onChange={[Function]}
        placeholder="LinkedIn URL"
        type="text"
      />
      <field
        label="Github:"
        name="githubusername"
        onChange={[Function]}
        placeholder="Github username only"
        type="text"
      />
    </div>
    <field
      label="Create"
      type="submit"
    />
    <field
      label="Cancel"
      onClick={[Function]}
      type="button"
    />
  </form>
</Overlay>
`;
